api/cas/views.py

```diff
@@ -0,0 +1,310 @@
+import json
+
+from rest_framework import generics
+from rest_framework import status
+from rest_framework.exceptions import ValidationError
+from rest_framework.response import Response
+
+from api.base.views import JSONAPIBaseView
+from api.cas import login, account, messages, permissions, service
+from api.cas.mixins import APICASMixin
+
+
+class LoginOsf(APICASMixin, generics.CreateAPIView, JSONAPIBaseView):
```

Comment:
Default OSF login view. Everything is handled into the `CasLoginAuthentication` class.